app/src/main/kotlin/app/youkai/ui/feature/settings/SettingsActivity.kt

```diff
@@ -0,0 +1,55 @@
+package app.youkai.ui.feature.settings
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Toast
+import app.youkai.BuildConfig
+import app.youkai.R
+import app.youkai.data.local.Credentials
+import com.hannesdorfmann.mosby.mvp.viewstate.MvpViewStateActivity
+import kotlinx.android.synthetic.main.activity_settings.*
+
+class SettingsActivity : MvpViewStateActivity<SettingsView, SettingsPresenter>(), SettingsView {
+
+    override fun createPresenter() = SettingsPresenter()
+
+    override fun createViewState() = SettingsState()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_settings)
+        setSupportActionBar(toolbar)
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+
+        checkPreferenceView.setOnClickListener {
+            checkPreferenceView.value = checkPreferenceView.value != true
```

Comment:
Can probably do `!checkPreferenceView.value` here (and [below](https://github.com/youkai-app/Youkai/pull/31/files#diff-4e2fd94a6ee99433fadff84f4e57a268R30)) ? 🙃

Reply:
This actually doesn't work because `value` is nullable... 😓

Reply:
Ah, makes sense. 😓